common/djangoapps/student/views.py

```diff
@@ -573,6 +573,28 @@ def is_course_blocked(request, redeemed_registration_codes, course_key):
     return blocked
 
 
+def compose_and_send_email(user, profile, registration=None, request=None):
+    """Compose an email content for user and send activation email"""
+    dest_addr = user.email
+    context = {
+        'name': profile.name,
+        'key': registration.activation_key,
```

Comment:
`registration` has a default value equals to `None`. If `registration` is not passed, then this line will break.